.github/workflows/create.yml

```diff
@@ -47,11 +59,23 @@ jobs:
           distribution: 'temurin'
           architecture: ${{ matrix.architecture }}
 
-      - name: Build wheels
+      - name: Build Windows and MacOS wheels
+        if: ${{ matrix.os != 'ubuntu-latest' }}
         env:
           CIBW_SKIP: '*musllinux* ${{ matrix.cibw_skip }}'
         run: python -m cibuildwheel --output-dir dist
 
+      - name: Build and test Linux wheels
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        env:
+          CIBW_SKIP: '*musllinux* ${{ matrix.cibw_skip }}'
+          CIBW_ARCHS: '${{ matrix.cibw_archs }}'
+          CIBW_TEST_REQUIRES: 'pytest'
+          CIBW_TEST_EXTRAS: 'dev,ci'
+          CIBW_BEFORE_TEST: 'yum install ant -y && cd {project} && ant all'
+          CIBW_TEST_COMMAND: 'cd {project}/tests/ && CLASSPATH=../build/test-classes:../build/classes python -m pytest -v'
+        run: python -m cibuildwheel --output-dir dist
```

Comment:
Can we keep the tests separate from the build phase?

Reply:
Previously, it was built in this step and tested later on in a separate step. We want to use that approach? This current approach is to build the wheels and if successful, the CIBW_TEST_COMMAND is run to test the wheels.

Reply:
Yep, but unfortunately (as shown here: https://github.com/pypa/cibuildwheel/blob/main/README.md#how-it-works), tests are run into the same container / runner session.
Keeping it separate allows us to be sure that the build scripts/installations are not influencing the tests.
(A new and clean `virtualenv` is created and used for tests, but it's not enough sometime)

Reply:
We can test the wheels in a separate step for x86_64. For aarch64, that won't be possible because github natively does not support arm64 architecture.

Reply:
For `aarch64` tests, `qemu` will help 😀

When setting up the test for `aarch64`, we can add:

```
- name: Set up QEMU for aarch64 tests
  if: ${{ matrix.test_arch == 'aarch64' }} # here we need to add `test_arch` in the matrix.
  uses: docker/setup-qemu-action@v1
  with:
      platforms: arm64
```

Reply:
@misl6 True, but that only sets up the QEMU for aarch64 binary. Any commands run further will be on the host architecture. To run the tests, we will have to use a docker image (manylinux2014_aarch64), setup python, pytest, ant. Then install the generated wheels, run 'ant all' and then the unit tests. cibuildwheels sets all of that up for us when we specify what we need using CIBW_TEST_REQUIRES. Unfortunately, cibuildwheel does not have a 'skip build and test only' option. We'll probably end up re-implementing the testing logic of cibuildwheel ourselves, unless there is another option.

Reply:
@misl6 Any suggestions?

Reply:
Hi @janaknat !
We're doing something similar here (that is actually used to build dependencies, as we can cache them, but the same approach could be used to process tests): https://github.com/kivy/kivy/blob/a94505ef854acae0046ab8d822a4c96c011bd269/.github/workflows/manylinux_wheels.yml#L73-L77